### ISSUE TITLE
Recovery Services - 2021-12-01

### ISSUE THREAD

**Issue by mbfrahry** [hashicorp/pandora#2345]
There is a [GetOperation](https://github.com/Azure/azure-sdk-for-go/blob/v65.0.0/services/recoveryservices/mgmt/2021-12-01/backup/protectioncontaineroperationresults.go#L42) function we are currently using in the Azure go sdk that I can't find Pandora.

I did see find it in the [swagger for this service](https://github.com/Azure/azure-rest-api-specs/blob/8a28143c7271d4496296ed47f70c3cb5a9981e57/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-12-01/bms.json#L3351) but I'm not sure if it's being mapped into pandora correctly

**Comment by tombuildsstuff** [comment]
@mbfrahry what are we using this for? The `OperationResult` would be to query the LRO, which should be happening automagically, so we shouldn't need to call this, providing we're polling on the LRO?

**Comment by mbfrahry** [comment]
It's for Inquiring when a new file share is added to an existing storage account. I haven't seen this before but here is the PR for context on what is going on https://github.com/hashicorp/terraform-provider-azurerm/commit/dd3769186e61e14b196b6ef56fde3c63fe4d7ead

**Comment by tombuildsstuff** [comment]
Right, in this case that operation is an LRO that should be marked as an LRO, then all of that logic can be removed in favour of an `InquireThenPoll` SDK call - I've opened https://github.com/Azure/azure-rest-api-specs/pull/23524 to fix the Swagger, hopefully that's fixed quickly but if not we can add a Swagger workaround for that until the upstream issue is fixed.

**Comment by tombuildsstuff** [comment]
I've updated the upstream PR to target API version `2023-06-01`, since there's a number of new API Versions since this opened
